tests/ops/functions/test_commutator.py

```diff
@@ -225,12 +235,12 @@ def test_comm_relations_pauli_words(self, op1, op2, true_res, transform_type1, t
         assert isinstance(res, Operator)
         assert isinstance(res, SProd)
 
-    def test_consistency_with_native_pauli_comms(self):
-        """Test consistent behavior between native comms in PauliWord and PauliSentence and qml.commutor"""
+    def test_paulis_used_when_ever_possible(self, mocker):
+        """Test that pauli_rep is used whenever possible even when ``pauli=False``"""
+        spy3 = mocker.spy(qml.QueuingManager, "stop_recording")
         op1 = qml.PauliX(0) @ qml.PauliX(1)
         op2 = qml.PauliY(0) + qml.PauliY(1)
-        res1 = qml.commutator(op1, op2, pauli=True)
-        res2 = PauliWord({0: "X", 1: "X"}).commutator(PauliWord({0: "Y"}) + PauliWord({1: "Y"}))
-        assert isinstance(res1, PauliSentence)
-        assert isinstance(res2, PauliSentence)
-        assert res1 == res2
+        res = qml.commutator(op1, op2, pauli=False)
+        spy3.assert_not_called()
```

Comment:
I dont understand why, but this is `False` 🤔

Reply:
I think instead of spying on `stop_recording`, maybe we could test the required queuing behavior with an `AnnotatedQueue()`?

Reply:
Just for clarity, I am here not testing that it is not queuing anything, but just that it never reaches the point of the stop_recording below, and instead stays in the "pauli branch" of the `qml.commutator` function even when `pauli=False`

Reply:
In particular, it uses pauli arithmetic whenever both ops have a `pauli_rep` available - even when `pauli=True`

Reply:
I found a different thing to test here (i.e. that `PauliSentence.operation` is called.

I am still curious why this doesnt work, probably something I am misunderstanding of the mocker.spy